test/gtest/sgd.hpp

```diff
@@ -0,0 +1,230 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include "cpu_sgd.hpp"
+#include "get_handle.hpp"
+#include "random.hpp"
+#include "tensor_holder.hpp"
+#include "verify.hpp"
+#include <gtest/gtest.h>
+#include <iostream>
+#include <miopen/sgd.hpp>
+#include <miopen/miopen.h>
+#include <vector>
+
+template <class T>
+inline std::ostream& operator<<(std::ostream& os, const std::vector<T>& v)
+{
+    os << '{';
+    for(int i = 0; i < v.size(); ++i)
+    {
+        if(i != 0)
+            os << ',';
+        os << v[i];
+    }
+    os << '}';
+    return os;
+}
+
+struct SGDTestCase
+{
+    std::vector<size_t> dims;
+    double lr;
+    double momentum;
+    double dampening;
+    double weightDecay;
+    bool nesterov;
+    bool momentumInitialized;
+    bool is_contiguous = true;
+
+    friend std::ostream& operator<<(std::ostream& os, const SGDTestCase& tc)
+    {
+        return os << " dims:" << tc.dims << " LearningRate:" << tc.lr << " Momentum:" << tc.momentum
+                  << " Dampening:" << tc.dampening << " WeightDecay:" << tc.weightDecay
+                  << " Nesterov:" << (int)tc.nesterov
+                  << " MomentumInitialized:" << (int)tc.momentumInitialized
+                  << "is_contiguous:" << tc.is_contiguous;
+    }
+
+    std::vector<size_t> GetDims() const { return dims; }
```

Comment:
It looks excessived.

Reply:
this is no excessived, return dims from test case to setup

Reply:
I see that you use it in auto dims    = SGD_config.GetDims();. You can use SGD_config.dims, like the way you do for other parameter in SGD_config ('lr`, 'momentum', ...)

Reply:
removed